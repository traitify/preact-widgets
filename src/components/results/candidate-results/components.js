export CandidateResults from "./index";
export PersonalityArchetype from "./personality-archetype";
export PersonalityDetails from "./personality-details";
export PersonalityDimensionColumn from "./personality-dimension-column";
export PersonalityDimensionDetails from "./personality-dimension-details";
export PersonalityDimensions from "./personality-dimensions";
