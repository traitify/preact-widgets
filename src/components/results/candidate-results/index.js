import {Component} from "react";
import PersonalityArchetype from "./personality-archetype";
import PersonalityDetails from "./personality-details";
import PersonalityDimensions from "./personality-dimensions";

export default class CandidateResults extends Component {
  render() {
    return (
      <section>
        <PersonalityArchetype {...this.props} />
        <PersonalityDimensions {...this.props} />
        <PersonalityDetails {...this.props} />
      </section>
    );
  }
}
