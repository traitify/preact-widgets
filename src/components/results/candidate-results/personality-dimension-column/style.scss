@import "~style/helpers";

.container {
  display: table-cell;
  padding: 60px 10px 30px 10px;
  position: relative;
  text-align: center;
  .icon {
    display: inline-block;
    height: 140px;
    padding: 25px;
    width: 140px;
    @include border-radius(50%);
    img {
      width: 100%;
    }
  }
  h2, h3 {
    font-weight: normal;
  }
  h2 {
    margin: 0;
    @include rems(26);
    @include mq(m) {
      @include rems(33);
    }
    @include mq(xl) {
      @include rems(36);
    }
  }
  h3 {
    @include rems(20);
    @include mq(m) {
      @include rems(24);
    }
    @include mq(xl) {
      @include rems(28);
    }
  }
  h3 {
    margin-bottom: 0;
  }
}
.background {
  position: absolute;
  left: 10px;
  right: 10px;
  bottom: 0;
  z-index: -1;
  &.low { top: 50%; }
  &.medium { top: 25%; }
  &.high { top: 0; }
}
