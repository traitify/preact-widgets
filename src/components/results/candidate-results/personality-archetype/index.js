import PropTypes from "prop-types";
import {Component} from "react";
import DangerousHTML from "lib/helpers/dangerous-html";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style";

class PersonalityArchetype extends Component {
  static defaultProps = {assessment: null, deck: null}
  static propTypes = {
    assessment: PropTypes.shape({archetype: PropTypes.object}),
    deck: PropTypes.shape({name: PropTypes.string.isRequired}),
    followDeck: PropTypes.func.isRequired,
    getOption: PropTypes.func.isRequired,
    isReady: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityArchetype.initialized", this);
    this.props.followDeck();
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityArchetype.updated", this);
  }
  render() {
    if(!this.props.isReady("results")) { return null; }
    if(!this.props.isReady("deck")) { return null; }

    const disabledComponents = this.props.getOption("disabledComponents") || [];
    if(disabledComponents.includes("PersonalityArchetype")) { return null; }

    const personality = this.props.assessment.archetype;
    if(!personality) { return null; }
    const badge = personality.details.find(({title}) => title === "Badge");
    const description = personality.details.find(({title}) => title === "Candidate Description");

    return (
      <div className={style.container}>
        <div className={style.details}>
          {badge && <img alt={personality.name} src={badge.body} />}
          <DangerousHTML
            html={this.props.translate("personality_heading", {
              deck_name: this.props.deck.name,
              personality: `<span>${personality.name}</span>`
            })}
            tag="h2"
          />
          {description && <p>{description.body}</p>}
        </div>
        <div className={style.meaning}>
          <DangerousHTML html={this.props.translate("candidate_description_for_archetype_html")} tag="p" />
        </div>
      </div>
    );
  }
}

export {PersonalityArchetype as Component};
export default withTraitify(PersonalityArchetype);
