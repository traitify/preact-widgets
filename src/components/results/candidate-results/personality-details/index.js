import PropTypes from "prop-types";
import {Component} from "react";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style";

class PersonalityDetails extends Component {
  static defaultProps = {assessment: null}
  static propTypes = {
    assessment: PropTypes.shape({archetype: PropTypes.object}),
    getOption: PropTypes.func.isRequired,
    isReady: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityDetails.initialized", this);
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityDetails.updated", this);
  }
  render() {
    if(!this.props.isReady("results")) { return null; }

    const personality = this.props.assessment.archetype || {};
    const {details} = personality;
    if(!details) { return null; }

    const disabledComponents = this.props.getOption("disabledComponents") || [];
    let disableSettings = disabledComponents.includes("PersonalitySettings");
    let disableTools = disabledComponents.includes("PersonalityTools");
    const settings = details.filter(({title}) => (title === "Settings that Work for You")).map(({body}) => body);
    const tools = details.filter(({title}) => (title === "Tools to Use")).map(({body}) => body);
    if(settings.length === 0) { disableSettings = true; }
    if(tools.length === 0) { disableTools = true; }
    if(disableSettings && disableTools) { return null; }

    return (
      <div className={style.container}>
        {!disableTools && (
          <div className={style.detail}>
            <div className={style.content}>
              <div className={style.bar} style={{background: "#008dc7"}} />
              <h4 className={style.title} style={{color: "#008dc7"}}>{this.props.translate("candidate_heading_for_tools")}</h4>
              <ul className={style.description}>
                {tools.map((tool) => (
                  <li key={tool}>{tool}</li>
                ))}
              </ul>
            </div>
          </div>
        )}
        {!disableSettings && (
          <div className={style.detail}>
            <div className={style.content}>
              <div className={style.bar} style={{background: "#32be4b"}} />
              <h4 className={style.title} style={{color: "#32be4b"}}>{this.props.translate("candidate_heading_for_settings")}</h4>
              <ul className={style.description}>
                {settings.map((setting) => (
                  <li key={setting}>{setting}</li>
                ))}
              </ul>
            </div>
          </div>
        )}
        <div className={style.clearfix} />
      </div>
    );
  }
}

export {PersonalityDetails as Component};
export default withTraitify(PersonalityDetails);
