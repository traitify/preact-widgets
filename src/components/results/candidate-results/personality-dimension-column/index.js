import PropTypes from "prop-types";
import {Component} from "react";
import {rgba} from "lib/helpers/color";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style.scss";

class PersonalityDimensionColumn extends Component {
  static propTypes = {
    translate: PropTypes.func.isRequired,
    type: PropTypes.shape({
      personality_type: PropTypes.shape({
        badge: PropTypes.shape({
          color_1: PropTypes.string.isRequired,
          image_medium: PropTypes.string.isRequired
        }).isRequired,
        level: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired
      }).isRequired,
      score: PropTypes.number.isRequired
    }).isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityDimensionColumn.initialized", this);
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityDimensionColumn.updated", this);
  }
  render() {
    const {translate, type: {personality_type: {badge, level, name}, score}} = this.props;
    const color = `#${badge.color_1}`;
    let backgroundClass;
    switch(true) {
      case score <= 3: backgroundClass = "low"; break;
      case score <= 6: backgroundClass = "medium"; break;
      default: backgroundClass = "high";
    }

    return (
      <li className={style.container}>
        <div>
          <div className={style.icon} style={{border: `3px solid ${color}`, background: rgba(color, 8.5)}}>
            <img src={badge.image_medium} alt={`${name} ${translate("badge")}`} />
          </div>
          <h3>{name}</h3>
          <h2>{level}</h2>
        </div>
        <div className={`${style.background} ${style[backgroundClass]}`} style={{background: rgba(color, 10)}} />
      </li>
    );
  }
}

export {PersonalityDimensionColumn as Component};
export default withTraitify(PersonalityDimensionColumn);
