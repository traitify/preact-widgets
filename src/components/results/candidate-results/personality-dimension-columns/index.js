import PropTypes from "prop-types";
import {Component} from "react";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import PersonalityDimensionColumn from "../personality-dimension-column";
import style from "./style";

class PersonalityDimensionColumns extends Component {
  static defaultProps = {assessment: null}
  static propTypes = {
    assessment: PropTypes.shape({personality_types: PropTypes.array}),
    isReady: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityDimensionColumns.initialized", this);
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityDimensionColumns.updated", this);
  }
  render() {
    if(!this.props.isReady("results")) { return null; }

    const types = this.props.assessment.personality_types.sort((x, y) => {
      const xDetail = x.personality_type.details.find(({title}) => title === "Position") || {};
      const yDetail = y.personality_type.details.find(({title}) => title === "Position") || {};

      return (xDetail.body || 1) - (yDetail.body || 1);
    });

    return (
      <div className={style.container}>
        <p>{this.props.translate("candidate_description_for_dimensions")}</p>
        <ul className={style.columns}>
          {types.map((type) => (
            <PersonalityDimensionColumn
              key={type.personality_type.id}
              type={type}
              {...this.props}
            />
          ))}
        </ul>
        <div className={style.columnsBuffer} />
      </div>
    );
  }
}

export {PersonalityDimensionColumns as Component};
export default withTraitify(PersonalityDimensionColumns);
