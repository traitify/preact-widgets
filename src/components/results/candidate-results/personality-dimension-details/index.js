import PropTypes from "prop-types";
import {Component} from "react";
import {detailWithPerspective} from "lib/helpers";
import {rgba} from "lib/helpers/color";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style";

class PersonalityDimensionDetails extends Component {
  static propTypes = {
    getOption: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired,
    type: PropTypes.shape({
      personality_type: PropTypes.shape({
        badge: PropTypes.shape({
          color_1: PropTypes.string.isRequired,
          image_medium: PropTypes.string.isRequired
        }).isRequired,
        details: PropTypes.array.isRequired,
        level: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired
      }).isRequired
    }).isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityDimensionDetails.initialized", this);
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityDimensionDetails.updated", this);
  }
  render() {
    const {translate, type: {personality_type: {badge, details, level, name}}} = this.props;
    const color = `#${badge.color_1}`;
    const benefits = details.filter(({title}) => (title === "Benefits")).map(({body}) => body);
    const perspective = this.props.getOption("perspective") || "firstPerson";
    const options = {base: {details}, perspective};

    return (
      <li className={style.container} style={{background: rgba(color, 10), borderTop: `5px solid ${color}`, listStyle: "none"}}>
        <div className={style.side}>
          <p className={style.icon}>
            <img src={badge.image_medium} alt={`${name} ${translate("badge")}`} />
          </p>
        </div>
        <div className={style.content}>
          <h2>{name} <span style={{color}}>|</span> {level}</h2>
          {perspective === "firstPerson" && (
            <h3>{translate("candidate_heading_for_dimension", {level, name})}</h3>
          )}
          <p className={style.description}>{detailWithPerspective({...options, name: "short_description"})}</p>
        </div>
        <h3>{translate("candidate_heading_for_benefits", {level, name})}</h3>
        <ul className={style.benefits}>
          {benefits.map((benefit) => (
            <li key={benefit} style={{background: rgba(color, 50)}}>{benefit}</li>
          ))}
        </ul>
      </li>
    );
  }
}

export {PersonalityDimensionDetails as Component};
export default withTraitify(PersonalityDimensionDetails);
