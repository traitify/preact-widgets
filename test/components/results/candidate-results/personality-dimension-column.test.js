import {Component} from "components/results/candidate-results/personality-dimension-column";
import ComponentHandler from "support/component-handler";
import assessment from "support/json/assessment/dimension-based.json";

jest.mock("lib/with-traitify", () => ((value) => value));

describe("PersonalityDimensionColumn", () => {
  let props;

  beforeEach(() => {
    props = {
      translate: jest.fn().mockName("translate"),
      type: assessment.personality_types[0],
      ui: {
        current: {},
        off: jest.fn().mockName("off"),
        on: jest.fn().mockName("on"),
        trigger: jest.fn().mockName("trigger")
      }
    };
  });

  describe("callbacks", () => {
    it("triggers initialization", () => {
      const component = new ComponentHandler(<Component {...props} />);

      expect(props.ui.trigger).toHaveBeenCalledWith("PersonalityDimensionColumn.initialized", component.instance);
    });

    it("triggers update", () => {
      const component = new ComponentHandler(<Component {...props} />);
      component.updateProps();

      expect(props.ui.trigger).toHaveBeenCalledWith("PersonalityDimensionColumn.updated", component.instance);
    });
  });

  it("renders component", () => {
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders component with low score", () => {
    props.type = {...props.type, score: 1};
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders component with medium score", () => {
    props.type = {...props.type, score: 5};
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders component with high score", () => {
    props.type = {...props.type, score: 10};
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });
});
