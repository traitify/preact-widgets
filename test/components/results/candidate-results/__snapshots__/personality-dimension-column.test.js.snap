// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalityDimensionColumn renders component 1`] = `
<li>
  <div>
    <div
      style={
        Object {
          "background": "rgba(41,183,112,0.085)",
          "border": "3px solid #29B770",
        }
      }
    >
      <img
        alt="Agreeableness undefined"
        src="https://cdn.traitify.com/badges/96dbf653-6ea0-4dc0-94dc-f6db3247510c_medium"
      />
    </div>
    <h3>
      Agreeableness
    </h3>
    <h2>
      Medium
    </h2>
  </div>
  <div
    className="undefined undefined"
    style={
      Object {
        "background": "rgba(41,183,112,0.1)",
      }
    }
  />
</li>
`;

exports[`PersonalityDimensionColumn renders component with high score 1`] = `
<li>
  <div>
    <div
      style={
        Object {
          "background": "rgba(41,183,112,0.085)",
          "border": "3px solid #29B770",
        }
      }
    >
      <img
        alt="Agreeableness undefined"
        src="https://cdn.traitify.com/badges/96dbf653-6ea0-4dc0-94dc-f6db3247510c_medium"
      />
    </div>
    <h3>
      Agreeableness
    </h3>
    <h2>
      Medium
    </h2>
  </div>
  <div
    className="undefined undefined"
    style={
      Object {
        "background": "rgba(41,183,112,0.1)",
      }
    }
  />
</li>
`;

exports[`PersonalityDimensionColumn renders component with low score 1`] = `
<li>
  <div>
    <div
      style={
        Object {
          "background": "rgba(41,183,112,0.085)",
          "border": "3px solid #29B770",
        }
      }
    >
      <img
        alt="Agreeableness undefined"
        src="https://cdn.traitify.com/badges/96dbf653-6ea0-4dc0-94dc-f6db3247510c_medium"
      />
    </div>
    <h3>
      Agreeableness
    </h3>
    <h2>
      Medium
    </h2>
  </div>
  <div
    className="undefined undefined"
    style={
      Object {
        "background": "rgba(41,183,112,0.1)",
      }
    }
  />
</li>
`;

exports[`PersonalityDimensionColumn renders component with medium score 1`] = `
<li>
  <div>
    <div
      style={
        Object {
          "background": "rgba(41,183,112,0.085)",
          "border": "3px solid #29B770",
        }
      }
    >
      <img
        alt="Agreeableness undefined"
        src="https://cdn.traitify.com/badges/96dbf653-6ea0-4dc0-94dc-f6db3247510c_medium"
      />
    </div>
    <h3>
      Agreeableness
    </h3>
    <h2>
      Medium
    </h2>
  </div>
  <div
    className="undefined undefined"
    style={
      Object {
        "background": "rgba(41,183,112,0.1)",
      }
    }
  />
</li>
`;
